### PR TITLE
migrations: Update memoryStore methods (for tests)

### PR DESCRIPTION
**Pull request by efritz**
Pulled from #29831. This PR updates the memoryStore used in tests touching the database to more accurately reflect the migration `Store` interface.

### CODE REVIEW

File: internal/database/connections/test/store.go [hunk 1]
```diff
@@ -34,7 +34,15 @@ func (s *memoryStore) Done(err error) error {
 }
 
 func (s *memoryStore) Version(ctx context.Context) (int, bool, bool, error) {
-	return s.version, s.dirty, s.versionSet, nil
+	if n := len(s.appliedVersions); n > 0 {
+		return s.pendingVersions[n-1], len(s.pendingVersions)+len(s.failedVersions) > 0, true, nil
```
Comment (camdencheek) [inline internal/database/connections/test/store.go:38]: is `pendingVersions` ever actually set? I only see reads, so it's a little concerning to me that we're indexing into it here

Comment (efritz) [reply internal/database/connections/test/store.go:38]: Check the file path :)

Comment (efritz) [reply internal/database/connections/test/store.go:38]: OH GOOD CATCH

Comment (efritz) [reply internal/database/connections/test/store.go:38]:
```suggestion
		return s.appliedVersions[n-1], len(s.pendingVersions)+len(s.failedVersions) > 0, true, nil
```

**Review by camdencheek — APPROVED** [review]

**Review by mrnugget — APPROVED** [review]

**Pull request merged**